corehq/apps/smsforms/app.py

```diff
@@ -100,38 +100,58 @@ def submit_unfinished_form(session):
 
     The form is only submitted if the smsforms session has not yet completed.
     """
-    # Get and clean the raw xml
+    formplayer_interface = FormplayerInterface(session.session_id, session.domain)
+    xml = _fetch_xml(formplayer_interface)
+
+    remove_case_actions = not session.include_case_updates_in_partial_submissions
+    cleaned_xml = _clean_xml_for_partial_submission(xml, should_remove_case_actions=remove_case_actions)
+
+    result = submit_form_locally(cleaned_xml, session.domain, app_id=session.app_id, partial_submission=True)
+    session.submission_id = result.xform.form_id
+
+
+def _fetch_xml(formplayer_interface):
+    """
+    :param formplayer_interface: FormplayerInterface obj
+    :return: xml if successful
+    """
     try:
-        response = FormplayerInterface(session.session_id, session.domain).get_raw_instance()
+        response = formplayer_interface.get_raw_instance()
         # Formplayer's ExceptionResponseBean includes the exception message,
-        # stautus ("error"), url, and type ("text")
+        # status ("error"), url, and type ("text")
         if response.get('status') == 'error':
             raise TouchformsError(response.get('exception'))
         xml = response['output']
     except InvalidSessionIdException:
         return
```

Comment:
Handling this exception here and returning `None` looks like this function is converting an exception to a error code. What do you think of moving the `try`/`except` up to the function that calls `_fetch_xml` (`submit_unfinished_form`) and then handle the error there? (I think by returning immediately when the error is caught)

It is unclear what would happen if `_clean_xml_for_partial_submission` were called with `None`, and I wasn't able to easily tell if that scenario is tested.

Did moving this `return` inside a function and not returning in the outer function change behavior? After re-reading this I'm wondering if this is broken?

Reply:
Yeah great catch! Moving up a level makes sense to me and I added a test to `_clean_xml_for_partial_submission` to ensure it will raise an exception if xml is missing. See fe6bbeb792caeca91716ed4572b450a2a9199f4f